core/src/test/java/org/apache/iceberg/TestSnapshotRefParser.java

```diff
@@ -0,0 +1,156 @@
+    String json = "{\"snapshot-id\":1,\"type\":\"tag\",\"max-ref-age-ms\":1}";
+    SnapshotRef ref = SnapshotRef.tagBuilder(1L)
+        .maxRefAgeMs(1L)
+        .build();
+    Assert.assertEquals("Should be able to deserialize tag with all fields", ref, SnapshotRefParser.fromJson(json));
+  }
+
+  @Test
+  public void testBranchFromJsonDefault() {
+    String json = "{\"snapshot-id\":1,\"type\":\"branch\"}";
+    SnapshotRef ref = SnapshotRef.branchBuilder(1L).build();
+    Assert.assertEquals("Should be able to deserialize default branch", ref, SnapshotRefParser.fromJson(json));
+  }
+
+  @Test
+  public void testBranchFromJsonAllFields() {
+    String json = "{\"snapshot-id\":1,\"type\":\"branch\",\"min-snapshots-to-keep\":2," +
+        "\"max-snapshot-age-ms\":3,\"max-ref-age-ms\":4}";
+    SnapshotRef ref = SnapshotRef.branchBuilder(1L)
+        .minSnapshotsToKeep(2)
+        .maxSnapshotAgeMs(3L)
+        .maxRefAgeMs(4L)
+        .build();
+    Assert.assertEquals("Should be able to deserialize branch with all fields", ref, SnapshotRefParser.fromJson(json));
+  }
+
+  @Test
+  public void testFailParsingWhenNullOrEmptyJson() {
+    String nullJson = null;
+    AssertHelpers.assertThrows("SnapshotRefParser should fail to deserialize null JSON string",
+        IllegalArgumentException.class, "Cannot parse snapshot ref from invalid JSON",
+        () ->  SnapshotRefParser.fromJson(nullJson));
+
+    String emptyJson = "";
+    AssertHelpers.assertThrows("SnapshotRefParser should fail to deserialize empty JSON string",
+        IllegalArgumentException.class, "Cannot parse snapshot ref from invalid JSON",
+        () ->  SnapshotRefParser.fromJson(emptyJson));
+  }
+
+  @Test
+  public void testFailParsingWhenMissingRequiredFields() {
+    String refMissingType = "{\"snapshot-id\":1}";
+    AssertHelpers.assertThrows("SnapshotRefParser should fail to deserialize ref with missing type",
+        IllegalArgumentException.class, "Cannot parse missing string",
+        () ->  SnapshotRefParser.fromJson(refMissingType));
+
+    String refMissingSnapshotId = "{\"type\":\"branch\"}";
+    AssertHelpers.assertThrows("SnapshotRefParser should fail to deserialize ref with missing snapshot id",
+        IllegalArgumentException.class, "Cannot parse missing long",
+        () ->  SnapshotRefParser.fromJson(refMissingSnapshotId));
+  }
+
+  @Test
+  public void testFailWhenFieldsHaveInvalidValues() {
+    String invalidSnapshotId = "{\"snapshot-id\":\"invalid-snapshot-id\",\"type\":\"not-a-valid-tag-type\"}";
+    AssertHelpers.assertThrows("SnapshotRefParser should fail to deserialize ref with invalid snapshot id",
+        IllegalArgumentException.class,
+        () ->  SnapshotRefParser.fromJson(invalidSnapshotId));
+
+    String invalidTagType = "{\"snapshot-id\":1,\"type\":\"not-a-valid-tag-type\"}";
+    AssertHelpers.assertThrows("SnapshotRefParser should fail to deserialize ref with invalid tag",
+        IllegalArgumentException.class,
```

Comment:
nit: would be good to verify the error msg here and above as well

Reply:
Good point, done!